### PR TITLE
Support Chakra.ICU on ARM

### PR DESCRIPTION
**Pull request by jackhorton**
This is the next major step in converting to ICU on all platforms. This change converts Chakra.ICU.Toolutil and Chakra.ICU.GenCCode to be built to run on the host architecture (usually x64) in order to generate icudtl_dat.obj, which is platform-agnostic and can be linked into an ARM-compatible DLL using the regular ARM linker. As a downside, it means that a clean build of Chakra.ICU takes about twice as long because ICU's common and i18n code are built twice, once for the target that gets linked to the resulting ChakraCore.dll, and once for the host that gets linked to GenCCode.exe. This is going to be a major problem for CI, however we can (hopefully) solve that problem later through caching

/cc @Microsoft/target-dev (I cant add target-dev as a reviewer for unknown reasons). I think this will also enable ARM64, however I haven't checked that yet.

### CODE REVIEW

File: deps/Chakra.ICU/Chakra.ICU.Common.vcxproj [hunk 1]
```diff
@@ -29,6 +29,7 @@
     </ClCompile>
     <Link Condition="'$(ChakraICU)'=='shared'">
       <SubSystem>Console</SubSystem>
+      <AdditionalDependencies>%(AdditionalDependencies);advapi32.lib</AdditionalDependencies>
```
Comment (jackhorton) [inline deps/Chakra.ICU/Chakra.ICU.Common.vcxproj]: This is for InitializeSecurityDescriptor, SetSecurityDescriptorDacl, and a handful of other functions. It seems to only actually be required for ARM, but common_uwp doesn't link against advapi32 on ARM. Are these APIs covered by WindowsApp.lib or one of the other dependencies in common_uwp?

**Comment by jackhorton** [comment]
Thanks to @Cellule for the idea of building the host-specific tools in a separate PowerShell script!

File: Build/Chakra.Build.props [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
     <ChakraICU Condition="'$(ChakraICU)'==''">false</ChakraICU>
     <!-- ARM does not support ICU until we figure out how to link icudtXXl.dat without genccode.exe -->
```
Comment (sethbrenith) [inline Build/Chakra.Build.props]: Comment looks not true anymore. Also we can remove the following line.

**Comment by jackhorton** [comment]
@MSLaguana suggested adding a flag to control building the data file, since for ARM in CI scenarios, we never actually use the data file (or anything, really, since those builds only build and don't run tests). I think I will update the build system with a new flag to control building the data file that we can turn off by default for certain CI scenarios.

**Comment by jackhorton** [comment]
Confirmation that the resulting binaries actually run on ARM:

image

File: deps/Chakra.ICU/Chakra.ICU.Common.vcxproj (hunk 1 above)
Comment (curtisman) [reply deps/Chakra.ICU/Chakra.ICU.Common.vcxproj]: Can you scope this dependencies only to ARM then?

**Comment by dilijev** [comment]
> /cc @Microsoft/target-dev (I cant add target-dev as a reviewer for unknown reasons).

I just added @Microsoft/target-dev as a Read-access contributor team to Microsoft/ChakraCore so you should be able to add that list as a reviewer now.

**Pull request merged**